src/atomate2/abinit/sets/base.py

```diff
@@ -0,0 +1,296 @@
+"""Module defining base abinit input set and generator."""
+import copy
+import logging
+import os
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Any, ClassVar, Iterable, List, Union
+
+import pseudo_dojo
+from abipy.abio.inputs import AbinitInput
+from monty.io import zopen
+from monty.json import MSONable
+from pymatgen.core.structure import Structure
+from pymatgen.io.abinit.pseudos import PseudoTable
+
+from atomate2.abinit.utils.common import (
+    INDATA_PREFIX,
+    INPUT_FILE_NAME,
+    OUTDATA_PREFIX,
+    TMPDATA_PREFIX,
+)
+from atomate2.common.sets import InputSet, InputSetGenerator
+
+__all__ = ["AbinitInputSet", "AbinitInputSetGenerator"]
+
+logger = logging.getLogger(__name__)
+
+
+class _UNSET(MSONable):
+    def __eq__(self, other):
+        return self.__class__ == other.__class__
+
+
+UNSET = _UNSET()
```

Comment:
Why can't you just use `None` to indicate unset? This is what we do in pymatgen.

Reply:
It's actually more complex than that. None can have a meaning to provide some default, e.g. we use None to say "use the recommended cut-offs for this composition". We may have a first generator where we explicitly set the ecut to a small value, e.g. 300 eV, then we explicitly set ecut=None to use the default automatic setting of the recommended cutoff. Anyway, I've removed the UNSET and used a different approach, where there is a register of the parameters that have been explicitly set. This may still need to be discussed and maybe another alternative approach could be used (did not think/find another one)...

Reply:
This feature has been removed in line of discussions we had and a simpler approach is now used.